src/types/utils.cpp

```diff
@@ -428,6 +428,108 @@ catch (...)
     return {};
 }
 
+// Routine Description:
+// - Pre-process text pasted (presumably from the clipboard) with provided option.
+// Arguments:
+// - wstr - String to process.
+// - option - option to use.
+// Return Value:
+// - The result string.
+std::wstring Utils::ConvertPasteString(const std::wstring& wstr, const PasteOption option)
+{
+    std::wstring converted;
+    converted.reserve(wstr.length());
+
+    const auto isControlCode = [](wchar_t c) {
+        if (c >= L'\x20' && c <= L'\x7f')
+        {
+            // Printable ASCII characters + DEL.
+            return false;
+        }
+
+        if (c > L'\x7f')
+        {
+            // Not a control character.
+            return false;
+        }
+
+        // All ASCII control characters will be removed except HT(0x09), LF(0x0a),
+        // CR(0x0d) and DEL(0x7F).
+        return c >= L'\x00' && c <= L'\x08' ||
+               c >= L'\x0b' && c <= L'\x0c' ||
+               c >= L'\x0e' && c <= L'\x1f';
+    };
+
+    std::wstring::size_type pos = 0;
+    std::wstring::size_type begin = 0;
+
+    if (WI_IsFlagSet(option, PasteOption::Bracketed))
+    {
+        converted.append(L"\x1b[200~");
+    }
+
+    while (pos < wstr.size())

```

Comment:
This is what I got for best performance. No more extra scaning.

I had to sacrifice extensibility for performance. Adding other options is still possible but also presumably harder than the lambda approach. As the saying goes, you win some you lose some.

Reply:
I do like this more. Thank you. :smile: